### PR TITLE
Speedup play #527

### PR DESCRIPTION
**Pull request by van800**
https://github.com/JetBrains/resharper-unity/issues/527

### CODE REVIEW

File: unity/JetBrains.Rider.Unity.Editor/EditorPlugin/PluginEntryPoint.cs [hunk 1]
```diff
@@ -105,8 +105,13 @@ private static void Init()
 
       InitializeEditorInstanceJson();
 
-      // for the case when files were changed and user just alt+tab to unity to make update, we want to fire
-      CsprojAssetPostprocessor.OnGeneratedCSProjectFiles();
+      // https://github.com/JetBrains/resharper-unity/issues/527#issuecomment-383326065
+      // process csproj files once per Unity process 
+      if (EditorPrefs.GetInt("Rider_UnityProcessId", 0) != Process.GetCurrentProcess().Id)
```
[the remark this answers is not in the document]
Comment (SugoiDev) [reply unity/JetBrains.Rider.Unity.Editor/EditorPlugin/PluginEntryPoint.cs]: ScriptableObjects are nice but might be overkill for that, no? I had a tremendous amount of trouble to get a similar system working properly across all kinds of reloads and partial reloads that Unity does. In particular when building and when (re)importing large amounts of data.

It's also kinda expensive to create and save them on every reload (saving a ScriptableObject is possibly more expensive than it is to process the csproj files).

[the remark this answers is not in the document]
Comment (citizenmatt) [reply unity/JetBrains.Rider.Unity.Editor/EditorPlugin/PluginEntryPoint.cs]: I don't know. It would need measuring, to be honest.

**Comment by van800** [comment]
Please review again.
I did some profiling - I do not see any perf degrade due to deserializing.

**Comment by van800** [comment]
@citizenmatt I need your review, please.

File: unity/JetBrains.Rider.Unity.Editor/EditorPlugin/Utils/RiderScriptableSingleton.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+using UnityEngine;
+
+namespace JetBrains.Rider.Unity.Editor.Utils
+{
+  [Location("JetBrainsRiderPluginCache.txt", LocationAttribute.Location.LibraryFolder)]
```
[the remark this answers is not in the document]
Comment (van800) [reply unity/JetBrains.Rider.Unity.Editor/EditorPlugin/Utils/RiderScriptableSingleton.cs]: I tested the scenario locally, it worked as expected. It wasn't deserializing value after Unity restart. and it it was deserializing on Play or other appdomain reload.

[the remark this answers is not in the document]
Comment (van800) [reply unity/JetBrains.Rider.Unity.Editor/EditorPlugin/Utils/RiderScriptableSingleton.cs]: However, I will try also without the cache file, thanks. Will follow up later.

**Comment by van800** [comment]
@citizenmatt I followed your advise and found a bug in my logic (
Fixed that and merged PR.

**Comment by SugoiDev** [comment]
For future reference, there's also [SessionState](https://docs.unity3d.com/2018.1/Documentation/ScriptReference/SessionState.html) that works like the EditorPrefs, but is only valid for the current session (survives assembly reload, but is reset when the editor is closed).

**Pull request merged**